[AMORO-3620][Improvement]: Fix the error in the calculation of the health score of table

cxxiii opened the pull request:
<!--
Thanks for sending a pull request!

Here are some tips for you:
  1. If this is your first time, please read our contributor guidelines: https://amoro.apache.org/how-to-contribute/
  2. If the PR is related to an issue in https://github.com/apache/amoro/issues, add '[AMORO-XXXX]' in your PR title, e.g., '[AMORO-XXXX] Your PR title ...'.
  3. If the PR is unfinished, add '[WIP]' in your PR title, e.g., '[WIP][AMORO-XXXX] Your PR title ...'.
-->

## Why are the changes needed?
<!--
Please clarify why the changes are needed. For instance,
  1. If you add a feature, you can talk about its use case.
  2. If you fix a bug, you can clarify why it is a bug.
  3. Use Fix/Resolve/Close #{ISSUE_NUMBER} to link this PR to its related issue
-->

Close #3620 .

## Brief change log
<!--
Clearly describe the changes made in modules, classes, methods, etc.
-->

- Fix the issue where the health score calculation is incorrect when the table is empty.
    - when the table is empty, which means the number of table files equals to 0, the table is considered to be healthy and its health score should be 100. However, the actual displayed score is 0. This PR revised this problem.

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

zhoujinsong commented on amoro-format-iceberg/src/main/java/org/apache/amoro/optimizing/plan/CommonPartitionEvaluator.java:
```diff
@@ -434,7 +434,7 @@ private double getTablePenaltyFactor(long dataFiles, long dataFilesSize) {
     // so there is only a small penalty
     return getNormalizedRatio(dataFiles, config.getMinorLeastFileCount())
-        * getNormalizedRatio(dataFilesSize, config.getTargetSize());
+        * getNormalizedRatio(dataFilesSize, minTargetSize);
```
Hi, can you give some reason why we should change here?

cxxiii replied:
I change here according to the computational formula of TablePenaltyFactor in https://github.com/apache/amoro/issues/3180 . I think there must have been a typing error before.

zhoujinsong commented:
@MarigWeizhi Can you help to review this PR?

MarigWeizhi commented:
1. I've tested locally and just need to modify one place: `avgHealthScore`, which should return 100 when partitionCount is 0. 
2. I'm not sure if it's necessary to change `targetSize` to `minTargetSize`

cxxiii commented:
> 1. I've tested locally and just need to modify one place: `avgHealthScore`, which should return 100 when partitionCount is 0.
> 2. I'm not sure if it's necessary to change `targetSize` to `minTargetSize`

1、I changed `targetSize` to `minTargetSize` according to the computational formula of TablePenaltyFactor in https://github.com/apache/amoro/issues/3180 . I think there must have been a typing error before.
2、I also changed the computational formula of averageDataFileSize according to the description provided in https://github.com/apache/amoro/issues/3180 . In the original code's calculation formula, the value of averageDataFileSize was incorrectly set to be less than or equal to 1. I made the necessary adjustments.

MarigWeizhi commented on amoro-format-iceberg/src/main/java/org/apache/amoro/optimizing/plan/CommonPartitionEvaluator.java:
```diff
@@ -397,7 +397,7 @@ public int getHealthScore() {
     long dataRecords = getFragmentFileRecords() + getSegmentFileRecords();
 
-    double averageDataFileSize = getNormalizedRatio(dataFilesSize, dataFiles);
+    double averageDataFileSize = dataFilesSize / dataFiles;
```
Is it possible for dataFiles to be 0 ？

MarigWeizhi replied:
> Is it possible for dataFiles to be 0 ？

I tested it locally, and when there is no data in the partition, this line of code is unreachable, so dataFiles will not be 0.

cxxiii replied:
I think it's impossible. The partition without dataFiles will not appear in the scan results, nor will the PartitionEvaluator for that partition be created.

MarigWeizhi commented:
> > 1. I've tested locally and just need to modify one place: `avgHealthScore`, which should return 100 when partitionCount is 0.
> > 2. I'm not sure if it's necessary to change `targetSize` to `minTargetSize`
> 
> 1、I changed `targetSize` to `minTargetSize` according to the computational formula of TablePenaltyFactor in #3180 . I think there must have been a typing error before. 2、I also changed the computational formula of averageDataFileSize according to the description provided in #3180 . In the original code's calculation formula, the value of averageDataFileSize was incorrectly set to be less than or equal to 1. I made the necessary adjustments.

LGTM
